test/mocks/MockDataRepository.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package mocks
+
+import base.BaseSpec
+import models.DynamicDataModel
+import org.scalamock.handlers.{CallHandler1, CallHandler2}
+import org.scalamock.scalatest.MockFactory
+import play.api.libs.json.Json.JsValueWrapper
+import reactivemongo.api.commands.{DefaultWriteResult, WriteError, WriteResult}
+import repositories.DynamicStubDataRepository
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait MockDataRepository extends BaseSpec with MockFactory {
+
+  val successWriteResult = DefaultWriteResult(ok = true, n = 1, writeErrors = Seq(), None, None, None)
```

Comment:
Missing some type annotations in here.

Reply:
IntelliJ is not showing any warnings for type annotations here. It typically doesn't warn you in certain situations when it's straightforward what type the `val` is to both human and compiler, so in this instance because the val is just being set to a Case Class, it's clear what the type is.